implementations/rust/ockam/ockam_identity/src/credential/identity.rs

```diff
@@ -81,15 +81,22 @@ impl<V: IdentityVault, S: AuthenticatedStorage> Identity<V, S> {
     }
 
     /// Present credential to other party, route shall use secure channel
-    pub async fn present_credential(&self, route: impl Into<Route>) -> Result<()> {
-        let credential = self.credential.read().await;
-        let credential = credential.as_ref().ok_or_else(|| {
-            Error::new(
-                Origin::Application,
-                Kind::Invalid,
-                "no credential to present",
-            )
-        })?;
+    pub async fn present_credential(
+        &self,
+        route: impl Into<Route>,
+        provided_credential: Option<&Credential>,
+    ) -> Result<()> {
+        let rw_credential = self.credential.read().await;
```

Comment:
Maybe we can refactor this code as `get_existing_credential_or(provided_credential)`? (to remove the duplication with [this part](https://github.com/build-trust/ockam/pull/4361/files#diff-cb334e90776a86359eca55c19120599a9f95fcc36a33633f8ccb0d541e976e19R130))